vault/identity_store_util.go

```diff
@@ -66,6 +100,16 @@ func (i *IdentityStore) loadGroups(ctx context.Context) error {
 				continue
 			}
 
+			// Ensure that there are no groups with duplicate names
+			groupByName, err := i.MemDBGroupByName(ctx, group.Name, false)
+			if err != nil {
+				return err
+			}
+			if groupByName != nil && !i.disableLowerCasedNames {
```

Comment:
What happens if a group is found and lower caes names *are* disabled? That's also an error condition but won't be covered here.

Reply:
When a group is found and when lower cased names are disabled, it means that we are falling back to the original behavior, and we are okay with the existence of another group. It is not an error condition. It would only mean that another group with same but differently cased name exists.

Reply:
It would be good to warn in that case just as with entity names -- that way they can try to address the issue.